src/main/java/it/smartcommunitylab/aac/saml/provider/SamlAuthenticationProvider.java

```diff
@@ -227,6 +240,18 @@ protected SamlUserAuthenticatedPrincipal createUserPrincipal(Object principal) {
         // we still don't have userId
         String userId = null;
 
+        // subjectId is optionally evaluated from an attribute defined in provider configurations
+        String subjectId = samlDetails.getName();
+        String subAttributeName = config.getConfigMap().getSubAttributeName();
+        if (StringUtils.hasText(subAttributeName)) {
+            try {
```

Comment:
just read the value if it's set, or fallback to name
```
String subjectId = StringUtils.hasText(subAttributeName) ? samlDetails.getFirstAttribute(subAttributeName) : samlDetails.getName();

```

if the attribute is missing subjectId will be null and the process will fail later.